src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -0,0 +1,152 @@
+﻿//
+// SearchBoxControl.cpp
+// Implementation of the SearchBoxControl class
+//
+
+#include "pch.h"
+#include "SearchBoxControl.h"
+#include "SearchBoxControl.g.cpp"
+
+using namespace winrt;
+using namespace winrt::Windows::UI::Xaml;
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    SearchBoxControl::SearchBoxControl() :
+        _goForward(false),
+        _isCaseSensitive(false)
+    {
+        InitializeComponent();
+
+        _goForwardButton = this->FindName(L"SetGoForwardButton").try_as<Controls::Primitives::ToggleButton>();
+        _goBackwardButton = this->FindName(L"SetGoBackwardButton").try_as<Controls::Primitives::ToggleButton>();
+
+        // TO DO: Is there a general way to put all the focusable elements in the
+        // collection ? Maybe try DFS
+        auto closeButton = this->FindName(L"CloseButton").try_as<Controls::Button>();

```

Comment:
This is .try_as. If they turn out to be null for whatever reason, does `.insert` below get upset? Should you fail gracefully if it doesn't cast correctly? Or is it a fail fast scenario for completely unexpected behavior that we programmed wrong?

Reply:
(Alternately, use .as so if it fails to cast, it'll crash right then)

Reply:
I would prefer to use try_as but add a check before I insert to the hashset. In this way, if the object turn out to be null, then at worst we just can not focus on it.

Reply:
That's fine with me.